src/algorithms/tracking/CKFTracking.cc

```diff
@@ -189,149 +195,232 @@ namespace eicrecon {
         // Set the CombinatorialKalmanFilter options
         CKFTracking::TrackFinderOptions options(
                 m_geoctx, m_fieldctx, m_calibctx, slAccessorDelegate,
-                extensions, Acts::LoggerWrapper{logger()}, pOptions, &(*pSurface));
+                extensions, pOptions, &(*pSurface));
+
+        // Create track container
+        auto trackContainer = std::make_shared<Acts::VectorTrackContainer>();
+        auto trackStateContainer = std::make_shared<Acts::VectorMultiTrajectory>();
+        // FIXME JANA2 std::vector<T*> requires wrapping TrackContainer, instead of:
+        //TrackContainer tracks(trackContainer, trackStateContainer);
+        std::vector<std::unique_ptr<ActsExamples::TrackContainer>> tracks_v;
+        tracks_v.push_back(
+          std::make_unique<ActsExamples::TrackContainer>(
+            trackContainer,
+            trackStateContainer));
+        auto& tracks = *(tracks_v.front().get());
+
+        // Add seed number column
+        tracks.addColumn<unsigned int>("seed");
+        Acts::TrackAccessor<unsigned int> seedNumber("seed");
+
+        // Loop over seeds
+        for (std::size_t iseed = 0; iseed < acts_init_trk_params.size(); ++iseed) {
+            auto result =
+                (*m_trackFinderFunc)(acts_init_trk_params.at(iseed), options, tracks);
 
-        auto results = (*m_trackFinderFunc)(acts_init_trk_params, options);
+            if (!result.ok()) {
+                m_log->debug("Track finding failed for seed {} with error {}", iseed, result.error());
+                continue;
+            }
 
-        for (std::size_t iseed = 0; iseed < acts_init_trk_params.size(); ++iseed) {
+            // Set seed number for all found tracks
+            auto& tracksForSeed = result.value();
+            for (auto& track : tracksForSeed) {
```

Comment:
Correct. `seedNumber` allows for grouping tracks by the seed they came from. With this upgrade the CKFTrackingFunction interface was changed to fit individually by seed (ActsExamples::TrackParameters), instead of passing all seeds at once (ActsExamples::TrackParametersContainer). That means that it is now up to us to keep track of the seed number, which we write with an extra accessor. The `tracksForSeed` is a container with references to entries in the overall TrackContainer that were obtained for this seed.